From: Jason Rhinelander <jason@imaginary.ca>
Date: Sun, 14 Jul 2019 17:36:16 -0300
Subject: randomx x86-64 build fix

randomx doesn't like ARCH_ID being set to a compiler subarchitecture, so
override it to x86-64 when compiled with the deb default `westmere`.
---
 external/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 7eed4c2..1999fee 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -46,6 +46,10 @@ sessiondep_or_submodule(liboxenc 1.5.0 oxen-encoding oxenc::oxenc)
 sessiondep_or_submodule(liboxenmq 1.3.2 oxen-mq oxenmq::oxenmq)
 
 add_subdirectory(db_drivers)
+
+if(ARCH_ID STREQUAL "westmere")
+  set(ARCH_ID "x86-64")  # deb workaround: randomx doesn't like ARCH_ID=westmere
+endif()
 add_subdirectory(randomx EXCLUDE_FROM_ALL)
 oxen_install_library(randomx)
 add_subdirectory(date EXCLUDE_FROM_ALL)
